File: C#/MatrixMultiplication.cs
using System;
using System.Diagnostics;

class Program
{
    static void Main()
    {
        Console.Write("Enter rows and columns of Matrix A: ");
        string[] partsA = Console.ReadLine().Split();
        int r1 = int.Parse(partsA[0]);
        int c1 = int.Parse(partsA[1]);

        Console.Write("Enter rows and columns of Matrix B: ");
        string[] partsB = Console.ReadLine().Split();
        int r2 = int.Parse(partsB[0]);
        int c2 = int.Parse(partsB[1]);

        if (c1 != r2)
        {
            Console.WriteLine("Matrix multiplication not possible (columns of A must equal rows of B).");
            return;
        }

        int[,] A = new int[r1, c1];
        int[,] B = new int[r2, c2];
        int[,] C = new int[r1, c2];

        Console.WriteLine("Enter elements of Matrix A row-wise:");
        for (int i = 0; i < r1; i++)
        {
            string[] row = Console.ReadLine().Split();
            for (int j = 0; j < c1; j++)
                A[i, j] = int.Parse(row[j]);
        }

        Console.WriteLine("Enter elements of Matrix B row-wise:");
        for (int i = 0; i < r2; i++)
        {
            string[] row = Console.ReadLine().Split();
            for (int j = 0; j < c2; j++)
                B[i, j] = int.Parse(row[j]);
        }

        Stopwatch sw = Stopwatch.StartNew();

        // Matrix multiplication
        for (int i = 0; i < r1; i++)
        {
            for (int k = 0; k < c1; k++)
            {
                int a = A[i, k];
                for (int j = 0; j < c2; j++)
                {
                    C[i, j] += a * B[k, j];
                }
            }
        }

        sw.Stop();

        Console.WriteLine("\nResultant Matrix (A x B):");
        for (int i = 0; i < r1; i++)
        {
            for (int j = 0; j < c2; j++)
                Console.Write(C[i, j] + " ");
            Console.WriteLine();
        }

        Console.WriteLine("\nExecution time: " + sw.Elapsed.TotalSeconds + " seconds");
    }
}

File: Rust/MatrixMultiplication.rs
use std::io::{self, Read};
use std::time::Instant;

fn main() {
    // Read all input at once
    let mut input = String::new();
    io::stdin().read_to_string(&mut input).unwrap();
    let mut it = input.split_whitespace();

    println!("Enter rows and columns of Matrix A:");
    let r1: usize = match it.next() {
        Some(v) => v.parse().unwrap(),
        None => return,
    };
    let c1: usize = it.next().unwrap().parse().unwrap();

    println!("Enter rows and columns of Matrix B:");
    let r2: usize = it.next().unwrap().parse().unwrap();
    let c2: usize = it.next().unwrap().parse().unwrap();

    if c1 != r2 {
        eprintln!("Matrix multiplication not possible (columns of A must equal rows of B).");
        return;
    }

    let mut a = vec![vec![0_i64; c1]; r1];
    let mut b = vec![vec![0_i64; c2]; r2];
    let mut c = vec![vec![0_i64; c2]; r1];

    println!("Enter elements of Matrix A (row-wise):");
    for i in 0..r1 {
        for j in 0..c1 {
            a[i][j] = it.next().unwrap().parse().unwrap();
        }
    }

    println!("Enter elements of Matrix B (row-wise):");
    for i in 0..r2 {
        for j in 0..c2 {
            b[i][j] = it.next().unwrap().parse().unwrap();
        }
    }

    let start = Instant::now();

    // Matrix multiplication (i-k-j order for better cache locality)
    for i in 0..r1 {
        for k in 0..c1 {
            let aik = a[i][k];
            for j in 0..c2 {
                c[i][j] += aik * b[k][j];
            }
        }
    }

    let duration = start.elapsed();

    println!("\nResultant Matrix (A x B):");
    for i in 0..r1 {
        for j in 0..c2 {
            print!("{} ", c[i][j]);
        }
        println!();
    }

    println!(
        "\nExecution time: {:.9} seconds",
        duration.as_secs_f64()
    );
}
